File: frontend/src/types/Bowler.ts
// Typescript definition for Bowler
export type Bowler = {
  bowlerId: number;
  bowlerLastName: string;
  bowlerFirstName: string;
  bowlerMiddleInit: string;
  team: {
    teamId: number;
    teamName: string;
  };
  bowlerAddress: string;
  bowlerCity: string;
  bowlerState: string;
  bowlerZip: string;
  bowlerPhoneNumber: string;
};

File: backend/IS 413 - Mission 10 - Backend/Data/Bowler.cs
// Model for Bowler

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace IS_413___Mission_10___Backend.Data
{
    public class Bowler
    {
        [Key]
        public int BowlerID { get; set; }
        public string? BowlerLastName { get; set; }
        public string? BowlerFirstName { get; set; }
        public string? BowlerMiddleInit { get; set; }
        // This is a foreign key that links to Team table
        [ForeignKey("TeamID")]
        public int? TeamID { get; set; }
        public Team? Team { get; set; }
        public string? BowlerAddress { get; set; }
        public string? BowlerCity { get; set; }
        public string? BowlerState { get; set; }
        public string? BowlerZip { get; set; }
        public string? BowlerPhoneNumber { get; set; }
    }
}
